src/python_testing/TC_SU_2_2.py

```diff
@@ -0,0 +1,844 @@
+#       --KVS /tmp/chip_kvs_requestor
+#       --trace-to json:${TRACE_APP}.json
+#     script-args: >
+#       --storage-path admin_storage.json
+#       --commissioning-method on-network
+#       --discriminator 1234
+#       --passcode 20202021
+#       --vendor-id 65521
+#       --product-id 32769
+#       --endpoint 0
+#       --trace-to json:${TRACE_TEST_JSON}.json
+#       --trace-to perfetto:${TRACE_TEST_PERFETTO}.perfetto
+#     factory-reset: true
+#     quiet: true
+# === END CI TEST ARGUMENTS ===
+
+import logging
+import asyncio
+import threading     # For Step #1
+import time          # For Step #2
+import subprocess
+import os
+import signal
+import psutil
+import socket
+
+
+import chip.clusters as Clusters
+from chip import ChipDeviceCtrl
+from chip.interaction_model import Status
+from chip.testing.matter_testing import MatterBaseTest, TestStep, async_test_body, default_matter_test_main, AttributeChangeCallback
+from chip.testing.matter_testing import ClusterAttributeChangeAccumulator, AttributeMatcher
+
+from mobly import asserts
+
+# Create a logger
+logger = logging.getLogger(__name__)
+
+
+class TC_SU_2_2(MatterBaseTest):
+    cluster_otap = Clusters.OtaSoftwareUpdateProvider
+    cluster_otar = Clusters.OtaSoftwareUpdateRequestor
+
+    async def launch_ota_provider(self, ota_file: str, provider_discriminator: int,
+                                  provider_setupPinCode: int, secured_device_port: int,
+                                  queue: str = None, timeout: int = None):
+        """
+        Launches the OTA Provider in a new macOS Terminal window.
+        Returns the subprocess.Popen object.
+        """
+        queue_cmd = f"-q {queue}" if queue else ""
+        timeout_cmd = f"-t {timeout}" if timeout else ""
+        cmd = f'''
+        osascript -e 'tell application "Terminal"
+            do script "cd /Users/jdelgado/connectedhomeip && \
+        source scripts/activate.sh && \
+        ./out/debug/chip-ota-provider-app --filepath {ota_file} \
+        --discriminator {provider_discriminator} \
+        --passcode {provider_setupPinCode} \
+        --secured-device-port {secured_device_port} {queue_cmd} {timeout_cmd}; exit"
+        end tell'
+        '''
+
+        proc = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        await asyncio.sleep(10)  # give the provider time to initialize
+        logger.info(f'Launched OTA Provider with command: {cmd}')
+        return proc
+
+    async def kill_ota_provider_process(self, process_name="chip-ota-provider-app", timeout=10):
+        """
+        Kills the OTA Provider process by name.
+        """
+        for proc in psutil.process_iter(['pid', 'name']):
+            if process_name in proc.info['name']:
+                logger.info(f'Killing OTA Provider process: {proc.info}')
+                proc.terminate()
+
+    async def kill_ota_requestor_process(self, process_name="ota.update", timeout=10):
```

Comment:
Since this is a server test, you should not be dealing with the ota requestor from inside the script. The DUT is the OTA requester.